File: style.php
<?php

$host = 'localhost'; // ou l'adresse du serveur

$dbname = 'library_db'; // nom de la base de données

$username = 'root'; // nom d'utilisateur

$password = ''; // mot de passe



try {

    // Création de la connexion à la base de données

    $pdo = new PDO("mysql:host=$host;dbname=$dbname", $username, $password);

    $pdo->setAttribute(PDO::ATTR_ERRMODE, PDO::ERRMODE_EXCEPTION);

} catch (PDOException $e) {

    echo 'La connexion a échoué : ' . $e->getMessage();

}

?>
<?php

include 'db_connection.php'; // Inclusion de la connexion à la base de données



if ($_SERVER['REQUEST_METHOD'] == 'POST') {

    $title = $_POST['title'];

    $author = $_POST['author'];

    $category = $_POST['category'];



    // Ajouter le livre à la base de données

    $query = "INSERT INTO books (title, author, category) VALUES (:title, :author, :category)";

    $stmt = $pdo->prepare($query);

    $stmt->bindParam(':title', $title);

    $stmt->bindParam(':author', $author);

    $stmt->bindParam(':category', $category);

    

    if ($stmt->execute()) {

        echo "Le livre a été ajouté avec succès!";

    } else {

        echo "Une erreur s'est produite lors de l'ajout du livre.";

    }

}

?>



<form action="add_book.php" method="POST">

    <label for="title">Titre du livre :</label><br>

    <input type="text" name="title" required><br><br>



    <label for="author">Auteur :</label><br>

    <input type="text" name="author" required><br><br>



    <label for="category">Catégorie :</label><br>

    <input type="text" name="category" required><br><br>



    <button type="submit">Ajouter le livre</button>

</form>

<?php

include 'db_connection.php'; // Inclusion de la connexion à la base de données



if ($_SERVER['REQUEST_METHOD'] == 'POST') {

    $title = $_POST['title'];

    $author = $_POST['author'];

    $category = $_POST['category'];



    // Ajouter le livre à la base de données

    $query = "INSERT INTO books (title, author, category) VALUES (:title, :author, :category)";

    $stmt = $pdo->prepare($query);

    $stmt->bindParam(':title', $title);

    $stmt->bindParam(':author', $author);

    $stmt->bindParam(':category', $category);

    

    if ($stmt->execute()) {

        echo "Le livre a été ajouté avec succès!";

    } else {

        echo "Une erreur s'est produite lors de l'ajout du livre.";

    }

}

?>



<form action="add_book.php" method="POST">

    <label for="title">Titre du livre :</label><br>

    <input type="text" name="title" required><br><br>



    <label for="author">Auteur :</label><br>

    <input type="text" name="author" required><br><br>



    <label for="category">Catégorie :</label><br>

    <input type="text" name="category" required><br><br>



    <button type="submit">Ajouter le livre</button>

</form>
?php

include 'db_connection.php'; // Inclusion de la connexion à la base de données



session_start();



if ($_SERVER['REQUEST_METHOD'] == 'POST') {

    $email = $_POST['email'];

    $password = $_POST['password'];



    // Vérification de l'utilisateur

    $query = "SELECT * FROM users WHERE email = :email";

    $stmt = $pdo->prepare($query);

    $stmt->bindParam(':email', $email);

    $stmt->execute();



    $user = $stmt->fetch(PDO::FETCH_ASSOC);



    if ($user && password_verify($password, $user['password'])) {

        // Si le mot de passe est correct

        $_SESSION['user_id'] = $user['id'];

        $_SESSION['user_email'] = $user['email'];

        header("Location: dashboard.php"); // Rediriger vers le tableau de bord

        exit;

    } else {

        echo "L'email ou le mot de passe est incorrect.";

    }

}

?>



<form action="login.php" method="POST">

    <label for="email">Email :</label><br>

    <input type="email" name="email" required><br><br>



    <label for="password">Mot de passe :</label><br>

    <input type="password" name="password" required><br><br>



    <button type="submit">Se connecter</button>

</form>

<?php

include 'db_connection.php'; // Inclusion de la connexion à la base de données



session_start();



// Vérifier si l'utilisateur est connecté

if (!isset($_SESSION['user_id'])) {

    header("Location: login.php"); // Si l'utilisateur n'est pas connecté, rediriger vers la page de connexion

    exit;

}



if (isset($_GET['book_id'])) {

    $book_id = $_GET['book_id'];

    $user_id = $_SESSION['user_id'];



    // Vérifier si le livre est disponible pour emprunt

    $query = "SELECT * FROM books WHERE id = :book_id AND available = 1";

    $stmt = $pdo->prepare($query);

    $stmt->bindParam(':book_id', $book_id);

    $stmt->execute();



    if ($stmt->rowCount() > 0) {

        // Si le livre est disponible, ajouter le prêt dans la table des prêts

        $query = "INSERT INTO loans (book_id, user_id, loan_date) VALUES (:book_id, :user_id, NOW())";

        $stmt = $pdo->prepare($query);

        $stmt->bindParam(':book_id', $book_id);

        $stmt->bindParam(':user_id', $user_id);



        if ($stmt->execute()) {

            // Mettre à jour la disponibilité du livre pour le marquer comme non disponible

            $query = "UPDATE books SET available = 0 WHERE id = :book_id";

            $stmt = $pdo->prepare($query);

            $stmt->bindParam(':book_id', $book_id);

            $stmt->execute();



            echo "Le livre a été emprunté avec succès!";

        } else {

            echo "Une erreur s'est produite lors de l'emprunt du livre.";

        }

    } else {

        echo "Le livre n'est pas disponible pour emprunt.";

    }

}

?>

File: style.java
import java.sql.Connection;

import java.sql.DriverManager;

import java.sql.SQLException;



public class DatabaseConnection {

    private static final String URL = "jdbc:mysql://localhost:3306/library_db";

    private static final String USER = "root";

    private static final String PASSWORD = "";

    

    public static Connection getConnection() throws SQLException {

        try {

            // Charger le driver JDBC

            Class.forName("com.mysql.cj.jdbc.Driver");

            // Retourner la connexion

            return DriverManager.getConnection(URL, USER, PASSWORD);

        } catch (ClassNotFoundException | SQLException e) {

            throw new SQLException("Erreur de connexion à la base de données", e);

        }

    }

}

import java.sql.Connection;

import java.sql.PreparedStatement;

import java.sql.SQLException;



public class AddBook {



    public static void addBook(String title, String author, String category) {

        String query = "INSERT INTO books (title, author, category) VALUES (?, ?, ?)";

        

        try (Connection connection = DatabaseConnection.getConnection();

             PreparedStatement statement = connection.prepareStatement(query)) {

            

            statement.setString(1, title);

            statement.setString(2, author);

            statement.setString(3, category);

            

            int rowsAffected = statement.executeUpdate();

            if (rowsAffected > 0) {

                System.out.println("Le livre a été ajouté avec succès !");

            } else {

                System.out.println("Une erreur s'est produite lors de l'ajout du livre.");

            }

            

        } catch (SQLException e) {

            e.printStackTrace();

        }

    }

}

import java.sql.*;



public class DisplayBooks {



    public static void displayBooks() {

        String query = "SELECT * FROM books";

        

        try (Connection connection = DatabaseConnection.getConnection();

             Statement statement = connection.createStatement();

             ResultSet resultSet = statement.executeQuery(query)) {

            

            System.out.println("Liste des livres :");

            while (resultSet.next()) {

                int id = resultSet.getInt("id");

                String title = resultSet.getString("title");

                String author = resultSet.getString("author");

                String category = resultSet.getString("category");

                

                System.out.println("ID: " + id + ", Titre: " + title + ", Auteur: " + author + ", Catégorie: " + category);

            }

            

        } catch (SQLException e) {

            e.printStackTrace();

        }

    }

}

import java.sql.*;



public class BorrowBook {



    public static void borrowBook(int userId, int bookId) {

        String query = "INSERT INTO loans (book_id, user_id, loan_date, return_date) VALUES (?, ?, NOW(), DATE_ADD(NOW(), INTERVAL 14 DAY))";

        

        try (Connection connection = DatabaseConnection.getConnection();

             PreparedStatement statement = connection.prepareStatement(query)) {

            

            statement.setInt(1, bookId);

            statement.setInt(2, userId);

            

            int rowsAffected = statement.executeUpdate();

            if (rowsAffected > 0) {

                // Mettre à jour la disponibilité du livre

                String updateQuery = "UPDATE books SET available = FALSE WHERE id = ?";

                try (PreparedStatement updateStatement = connection.prepareStatement(updateQuery)) {

                    updateStatement.setInt(1, bookId);

                    updateStatement.executeUpdate();

                }

                System.out.println("Le livre a été emprunté avec succès !");

            } else {

                System.out.println("Le livre n'a pas pu être emprunté.");

            }

            

        } catch (SQLException e) {

            e.printStackTrace();

        }

    }

}

import java.util.Scanner;



public class Main {



    public static void main(String[] args) {

        Scanner scanner = new Scanner(System.in);



        while (true) {

            System.out.println("\nGestion des livres et emprunts - Bibliothèque");

            System.out.println("1. Afficher les livres");

            System.out.println("2. Ajouter un livre");

            System.out.println("3. Emprunter un livre");

            System.out.println("4. Quitter");

            System.out.print("Choisissez une option : ");

            

            int choice = scanner.nextInt();

            scanner.nextLine();  // Consomme la nouvelle ligne



            switch (choice) {

                case 1:

                    DisplayBooks.displayBooks();

                    break;

                case 2:

                    System.out.print("Entrez le titre du livre : ");

                    String title = scanner.nextLine();

                    System.out.print("Entrez l'auteur du livre : ");

                    String author = scanner.nextLine();

                    System.out.print("Entrez la catégorie du livre : ");

                    String category = scanner.nextLine();

                    AddBook.addBook(title, author, category);

                    break;

                case 3:

                    System.out.print("Entrez votre ID utilisateur : ");

                    int userId = scanner.nextInt();

                    System.out.print("Entrez l'ID du livre à emprunter : ");

                    int bookId = scanner.nextInt();

                    BorrowBook.borrowBook(userId, bookId);

                    break;

                case 4:

                    System.out.println("Au revoir!");

                    return;

                default:

                    System.out.println("Option invalide. Essayez à nouveau.");

            }

        }

    }

}

<dependency>

    <groupId>mysql</groupId>

    <artifactId>mysql-connector-java</artifactId>

    <version>8.0.26</version> <!-- Vérifiez la dernière version disponible -->

</dependency>


